src/symbolize/gimli/elf.rs

```diff
@@ -43,6 +43,44 @@ impl Mapping {
         })
     }
 
+    /// On Android, shared objects can be loaded directly from a
+    /// ZIP archive. For example, an app may load a library from
+    /// `/data/app/com.example/base.apk!/lib/x86_64/mylib.so`
+    ///
+    /// For one of these "ZIP-embedded" libraries, `zip_offset` will be
+    /// non-zero (see [super::libs_dl_iterate_phdr]).
+    #[cfg(target_os = "android")]
+    pub fn new_android(path: &Path, zip_offset: usize) -> Option<Mapping> {
```

Comment:
Unless the zip offset is not a file offset, my comments about Mmap apply here as well. These are private types, despite the `pub` annotation.

Reply:
Just to clarify, do you want me to merge the two methods, or just change the type of `zip_offset`?

Reply:
both. specifically it's not clear to me why we should have `new_android` instead of just "`Mapping::new` takes an offset now".

Reply:
My reasoning was that the offset-handling code is Android-specific: the offset being non-zero signals that the file path should be treated as an archive on Android, and the offset wouldn't be used on any other platform.

If the methods are to be merged, I'm not sure what the expected behaviour should be if the offset is non-zero on other platforms - just ignore it?

Reply:
Oh...so zero is a magic number?

Is it literally impossible for us to want to map a zip file from offset 0?

Reply:
Especially considering the possibility of 16KiB pages?

Reply:
As far as I understand, not really? [zipalign](https://developer.android.com/tools/zipalign) will align these libraries such that the start of each file is page-aligned, and the dynamic linker will fail to load the library if it's not aligned correctly. And given that the ZIP header takes up some space at the start of the archive, I don't see how an embedded library on Android would have an offset of 0.

That said, I can wrap the offset in an `Option` if you'd prefer? (if for nothing else than clarity)

Reply:
Ah, zipalign was the missing piece of the puzzle for me, thank you. "Someone decided to shove their binary and its debuginfo inside a compressed archive" is just a Tuesday, but the specifics beyond that vary enough from case to case I didn't want to assume.

Yes, I would prefer the Option approach, and for Mapping it can remain under a `cfg` and leaving the other signatures intact.

I still would prefer Mmap was uniform though.